crates/ruff_python_formatter/src/other/f_string_element.rs

```diff
@@ -205,7 +250,11 @@ impl Format<PyFormatContext<'_>> for FormatFStringExpressionElement<'_> {
                     trailing_comments(comments.trailing(self.element)).fmt(f)?;
                 }
 
-                bracket_spacing.fmt(f)
+                if conversion.is_none() && format_spec.is_none() {
+                    bracket_spacing.fmt(f)?;
+                }
+
+                Ok(())
```

Comment:
This is an additional change which isn't relevant to this PR but should be a follow-up. I've included it in here for simplicity. For example,

```py
x = f"aaaaaaaaaaaaaaa {
    {'aaaaaaaaaaaaaaaaaa', 'dddddddddddddddddd'}!s
}"
```

This should be formatted as:
```py
x = f"aaaaaaaaaaaaaaa { {'aaaaaaaaaaaaaaaaaa', 'dddddddddddddddddd'}!s}"
```

Instead of:
```py
x = f"aaaaaaaaaaaaaaa { {'aaaaaaaaaaaaaaaaaa', 'dddddddddddddddddd'}!s }"
#                                                                     ^ this isn't required if there's format spec or conversion spec
```